### PR TITLE
SL-396 npm release scripts

### PR DESCRIPTION
**Pull request by chris-miaskowski**
**Reviewer notes**

I have copied changes from https://github.com/stoplightio/prism/pull/173 that was never merged.
It was part of https://stoplightio.atlassian.net/browse/SL-396 which hasn't been finished (was closed as "won't fix") because (AFAIR) we weren't sure we should release prism to git.

Anyway, we don't have much choice but to run those release scripts and release prism packages.

Since this is a monorepo I expect (I wasn't able to verify it) that once we run `npm run release.wet` we there will be four packages released and tagged
- @stoplight/prism-http-v1.0.0
- @stoplight/prism-http-server-v1.0.0
- @stoplight/prism-cli-v1.0.0
- @stoplight/prism-core-v1.0.0

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,5 +51,42 @@
...
+  "husky": {
+    "hooks": {
+      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
+    }
+  },
+  "release": {
+    "branch": "chore/sl-83/prism-packaging-and-binary-deployment",
+    "verifyConditions": [
+      "@semantic-release/git",
+      "@semantic-release/github",
```
Comment (marbemac) [inline package.json]: Remove

File: package.json [hunk 2]
```diff
@@ -40,5 +51,42 @@
...
+      "@commitlint/config-conventional"
+    ]
+  },
+  "husky": {
+    "hooks": {
+      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
+    }
+  },
+  "release": {
+    "branch": "chore/sl-83/prism-packaging-and-binary-deployment",
```
Comment (marbemac) [inline package.json]: `develop`?

File: package.json [hunk 3]
```diff
@@ -40,5 +51,42 @@
...
+    "verifyConditions": [
+      "@semantic-release/git",
+      "@semantic-release/github",
+      "@semantic-release/npm"
+    ],
+    "prepare": [
+      "@semantic-release/git",
```
Comment (marbemac) [inline package.json]: Remove. in fact, not sure we need `prepare`, `publish`, `success`, or `fail`?

File: package.json (hunk 1 above)
Comment (marbemac) [reply package.json]: Do we need `@semantic-release/commit-analyzer`?

File: package.json (hunk 3 above)
Comment (chris-miaskowski) [reply package.json]: Oh yeah I have that commit locally but forgot to push :) one second!

File: package.json (hunk 2 above)
Comment (chris-miaskowski) [reply package.json]: I use this for testing locally, will change to `develop`

File: package.json (hunk 1 above)
Comment (chris-miaskowski) [reply package.json]: I reckon we do for stuff like "BREAKING CHANGE" etc.

Comment (marbemac) [reply package.json]: Looks like might be already included actually? https://github.com/Updater/semantic-release-monorepo#release-config - search that section for "set of plugins it wraps".

Comment (chris-miaskowski) [reply package.json]: Let me check, I know recall that the `monorepo-semantic-release` re-wrapped plugins and maybe that's why.

File: package.json [hunk 4]
```diff
@@ -16,17 +16,28 @@
     "test.fast": "jest --env=node --maxWorkers=2",
     "test.coverage": "yarn test --coverage",
     "test.watch": "yarn test --watchAll",
-    "test.update": "yarn test.fast --updateSnapshot"
+    "test.update": "yarn test.fast --updateSnapshot",
+    "release.dry": "lerna exec -- semantic-release --dry-run -e semantic-release-monorepo --tag-format='${LERNA_PACKAGE_NAME}-v\\${version}'",
+    "release.wet": "lerna exec -- semantic-release -e semantic-release-monorepo --tag-format='${LERNA_PACKAGE_NAME}-v\\${version}'"
   },
   "devDependencies": {
+    "@commitlint/cli": "7.2.x",
```
Comment (XVincentX) [inline package.json]: I do not think we need this package

File: package.json [hunk 5]
```diff
@@ -40,5 +51,34 @@
...
+    }
+  },
+  "release": {
+    "branch": "develop",
+    "verifyConditions": [
+      "@semantic-release/git",
+      "@semantic-release/npm"
+    ],
+    "prepare": [
+      "@semantic-release/git",
```
Comment (XVincentX) [inline package.json]: Would it be possible to move all this stuff in `.releaserc`? That'd help us to keep `package.json` clean

File: .circleci/config.yml [hunk 6]
```diff
@@ -37,8 +37,23 @@ jobs:
           name: Run build
           command: yarn build
 
+    release:
+      docker:
+        - image: circleci/node:8
+      steps:
+        - checkout
+        - run: yarn
+        - run: yarn build
```
Comment (XVincentX) [inline .circleci/config.yml]: In theory `build` should be a `pre-release` task as well

File: lerna.json [hunk 7]
```diff
@@ -0,0 +1,4 @@
+{
+  "packages": ["packages/*"],
+  "version": "0.0.0"
+}
```
Comment (XVincentX) [inline lerna.json]: Please add `"useWorkspaces": true` so that lerna will leverage yarn workspaces.

File: lerna.json [hunk 8]
```diff
@@ -0,0 +1,4 @@
+{
+  "packages": ["packages/*"],
+  "version": "0.0.0"
```
Comment (XVincentX) [inline lerna.json]: Make sure to set yarn for npm client:

`"npmClient": "yarn",`

**Pull request closed without merging**